Treeview: Fixed focus bug (404) and  state using CSS content being read bug (406)

jongund opened the pull request.

mcking65 commented on examples/treeview/treeview-2/css/treeLinks.css:
```diff
@@ -23,29 +23,39 @@ ul[role="tree"] a {
+}
+
+[role="treeitem"][aria-expanded="false"] > span > span.state:before {
+  content: "\00229E \0000A0";
```
Still here; still causing issue 406

mcking65 commented on examples/treeview/treeview-2/treeview-2a.html:
```diff
@@ -55,7 +55,7 @@ <h3 id="tree1">Foods</h3>
             <li role="treeitem"
                 aria-expanded="false">
-              <span>Fruits</span>
+              <span><span class="state" aria-hidden="true"></span>Fruits</span>
```
the aria-hidden will not apply to a pseudo element

jongund commented:
Matt,

I reset my fork of the practices and made the following changes to the treeview examples:


  *   Removed default keyboard focus style from treeitems
  *   Changed the use of character entities to images in treeview-2 example

Looks like my pull request should now be clean.

Jon


From: Matt King <notifications@github.com>
Reply-To: w3c/aria-practices <reply@reply.github.com>
Date: Wednesday, June 21, 2017 at 4:39 AM
To: w3c/aria-practices <aria-practices@noreply.github.com>
Cc: Jon Gunderson <jongund@illinois.edu>, Author <author@noreply.github.com>
Subject: Re: [w3c/aria-practices] Treeview: Fixed focus bug (404) and state using CSS content being read bug (406) (#407)


@mcking65 requested changes on this pull request.

May be better to close this PR and start over. Please first delete your old treeview branch and make a fresh one that is built off the latest master so the pull request is clean.

This PR has 2 jscs errors.

It looks like one part of issue 404 is addressed; default focus is disabled. However, the custom focus is still implemented using js events rather than just css.

It does not fix issue 406 because aria-hidden will not apply to a pseudo element. Or, at least it doesn't in all browsers. It isn't clear to me that it should apply.

________________________________

In examples/treeview/treeview-2/css/treeLinks.css<https://github.com/w3c/aria-practices/pull/407#discussion_r123197384>:

>    color: #630;

   font-weight: bold;

   font-size: 1em;

+  position: relative;

+  top: -0.1em;

+}

+

+[role="treeitem"][aria-expanded="false"] > span > span.state:before {

+  content: "\00229E \0000A0";

Still here; still causing issue 406

________________________________

In examples/treeview/treeview-2/treeview-2a.html<https://github.com/w3c/aria-practices/pull/407#discussion_r123197885>:

> @@ -55,7 +55,7 @@ <h3 id="tree1">Foods</h3>

           <ul role="tree" aria-labelledby="tree1" >

             <li role="treeitem"

                 aria-expanded="false">

-              <span>Fruits</span>

+              <span><span class="state" aria-hidden="true"></span>Fruits</span>

the aria-hidden will not apply to a pseudo element

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/w3c/aria-practices/pull/407#pullrequestreview-45375238>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ABcE7HYohig1VHSWnbWxIO4WqdTv9REPks5sGOS5gaJpZM4N-cq3>.

mcking65 commented:
Replaced by pr #411. Closing.
